api/auth/index.ts

```diff
@@ -165,17 +179,200 @@ export async function oauth(req: express.Request, res: express.Response, next: e
+
+  if (!app.locals.issuer) {
+      try {
+          app.locals.issuer = await configureIssuer(idamWebUrl)
+          logger._logger.info('Issuer configured:', app.locals.issuer)
+      } catch (error) {
+          return next(error)
+      }
+  }
+
+  if (!app.locals.client) {
+      const clientMetadata: ClientMetadata = {
+          client_id: idamClient,
+          client_secret: secret,
+          response_types: ['code'],
+          token_endpoint_auth_method: 'client_secret_post', // The default is 'client_secret_basic'.
+      }
+
+      app.locals.client = new app.locals.issuer.Client(clientMetadata)
+      logger._logger.info('Client configured:', app.locals.client)
+  }
+
+  const host = req.get('host')
+  if (host.indexOf(':') > 0 ) {
+      const hostwithoutPort = host.substring(0, host.indexOf(':'))
+      if (net.isIP(hostwithoutPort)) {
+          return next()
+      }
+  }
+
+  console.log('host is', host)
+  let redirectUri = path.join(host, getConfigValue(OAUTH_CALLBACK_URL))
+  redirectUri = getConfigValue(PROTOCOL) + '://' + redirectUri
+  console.log('redirectUri', redirectUri)
+
+  // logger.info('configuring strategy with redirect_uri:', redirectUri)
+
+  passport.unuse('oidc').use('oidc', new Strategy({
+      client: app.locals.client,
+      params: {
+          prompt: 'login',
+          redirect_uri: redirectUri,
+          scope: 'profile openid roles manage-user create-user',
+      },
+      sessionKey: 'xui_webapp', // being explicit here so we can set manually on logout
+      usePKCE: false, // issuer doesn't support pkce - no code_challenge_methods_supported
+  }, oidcVerify))
+
+  next()
+}
+
+export async function openIdConnectAuth(req: express.Request, res: express.Response, next: express.NextFunction) {
+  passport.authenticate('oidc', (error, user, info) => {
+
+    // TODO: give a more meaningful error to user rather than redirect back to idam
+    // return next(error) would pass off to error.handler.ts to show users a proper error page etc
+    if (error) {
+      logger.error(error)
+      // return next(error)
+    }
+    if (info) {
+      logger.info(info)
+      // return next(info)
+    }
+    if (!user) {
+      logger._logger.info('No user found, redirecting')
+      return res.redirect('/auth/login')
+    }
+    req.logIn(user, err => {
+      if (err) {
+        return next(err)
+      }
+      logger._logger.info('logged in user, redirecting to authCallbackSuccess')
+      return openIdAuthCallbackSuccess(req, res)
+    })
+  })(req, res, next)
+}
+
+export function openIdAuthCallbackSuccess(req: express.Request, res: express.Response) {
+  logger._logger.info('authCallbackSuccess', req.session)
+
+  // we need extra logic before success redirect
+  const userDetails = req.session.passport.user
+  const roles = userDetails.userinfo.roles
+
+  axios.defaults.headers.common.Authorization = `Bearer ${userDetails.tokenset.access_token}`
```

Comment:
this was removed from axios, please do not reimplement it as it's a security issue

Reply:
this will be a different PR as discussed with @dyanesh